straitlets/builtin_models.py

```diff
@@ -58,11 +59,27 @@ def netloc(self):
         host_port = join_filter_empty(':', self.hostname, self.port)
         return '@'.join([user_pass, host_port])
 
+    query = Dict(
+        allow_none=True,
+        default_value=None,
```

Comment:
I'd probably have a slight preference for calling this `query_params` to be clear that it's the dict rather than the formatted string.